docs/chapters/baseplatform/replacing-packagegroups.rst

```diff
@@ -0,0 +1,63 @@
+:orphan:
+
+.. _replacing-packagegroups:
+
+Replacing packagegroups by hand-picked components
+==================================================
+
+By default PELUX include components from Qt Auto using the packagegroups feature.
+This chapter describes how to replace these default Qt Auto components with
+hand-picked components in a new distribution.
+
+The following packagegroups from Qt Auto are included in the PELUX image.
+
+* ``packagegroup-b2qt-automotive-addons``
+* ``nativesdk-packagegroup-b2qt-automotive-qt5-toolchain-host``
+* ``packagegroup-b2qt-automotive-qt5-toolchain-target``
+ 
+These packagegroups add several Qt Auto add-ons (includes ``neptune-ui`` among others) 
+and packages that are used when building SDK for host and target machines.
+
+If all of the Qt Auto components defined in above packagegroups are needed in the new 
+distribution then image recipe ``meta-<distro>/<image-recipe-path>/<image-name>.bb`` of 
+the new distribution can simply include ``core-image-pelux-qtauto-neptune`` as follows.
+
+.. code-block:: bash
+
+    DESCRIPTION = "New Distro image with QtAuto frontend"
+
+    require core-image-pelux-qtauto-neptune
+
+If the new distribution want to omit all or some of the Qt Auto components in the above
+mentioned packagegroups then it can simply remove the unwanted packagegroups from it's image
+recipe ``meta-<distro>/<image-recipe-path>/<image-name>.bb``. The example below shows
+how to omit packagegroups for building SDK for host and target machines. 
+
+.. code-block:: bash
+
+    DESCRIPTION = "New Distro image with QtAuto frontend"
+
+    require core-image-pelux-qtauto-neptune
+
+    TOOLCHAIN_HOST_TASK_remove = " nativesdk-packagegroup-b2qt-automotive-qt5-toolchain-host "
+    TOOLCHAIN_TARGET_TASK_remove = " packagegroup-b2qt-automotive-qt5-toolchain-target "
+
+Now if the new distribution also wants to omit the Qt Auto add-ons but still want to keep
+the ``neptune-ui`` or add some other new component ``<other-component>`` it can simply add
+the following to ``meta-<distro>/<image-recipe-path>/<image-name>.bb``. It can also do the
+same for host and target SDK build packages.
```

Comment:
In this case, wouldn't it be easier to just inherit `core-image-pelux-qtauto` and not having to `IMAGE_INSTALL_remove` anything? Or, if one wants neptune-ui and nothing else of the qt stuff, one can inherit `core-image-pelux` and just add neptune-ui?

Reply:
Yes I though so too, but that is a specific case and then it doesn't include the information about how to replace packagegroups which is the main purpose of this chapter.

Reply:
But there is no need to bend over backwards just to describe a case which one wouldn't use anyway, is it?